# HG changeset patch
# User Mike Miller <mtmiller@octave.org>
# Date 1456268904 28800
#      Tue Feb 23 15:08:24 2016 -0800
# Node ID 64be7d374f86f3baa61eb1d01a7144dd6e04de62
# Parent  5169d5993d77e460897536976197d7b05d1df700
Make OCTAVE_DEPRECATED macro compatible with older gcc

* mk-octave-config-h.sh, oct-conf-post.in.h (OCTAVE_DEPRECATED):
Apply msg argument only if compiling with a gcc version that accepts it.

diff --git a/build-aux/mk-octave-config-h.sh b/build-aux/mk-octave-config-h.sh
--- a/build-aux/mk-octave-config-h.sh
+++ b/build-aux/mk-octave-config-h.sh
@@ -90,17 +90,21 @@ since all of Octave's header files alrea
 #if ! defined (octave_octave_config_h)
 #define octave_octave_config_h 1
 
 #if ! defined (OCTAVE_AUTOCONFIG_H_INCLUDED)
 
 #include <stdint.h>
 
 #if defined (__GNUC__)
-#  define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__ (msg)))
+#  if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)
+#    define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__ (msg)))
+#  else
+#    define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__))
+#  endif
 #  define OCTAVE_NORETURN __attribute__ ((__noreturn__))
 #  define OCTAVE_UNUSED __attribute__ ((__unused__))
 
 #  define HAVE_OCTAVE_DEPRECATED_ATTR 1
 #  define HAVE_OCTAVE_NORETURN_ATTR 1
 #  define HAVE_OCTAVE_UNUSED_ATTR 1
 #else
 #  define OCTAVE_DEPRECATED(msg)
diff --git a/oct-conf-post.in.h b/oct-conf-post.in.h
--- a/oct-conf-post.in.h
+++ b/oct-conf-post.in.h
@@ -31,17 +31,21 @@ along with Octave; see the file COPYING.
    OCTAVE_ATTRIBUTE_NAME in place of vendor specific attribute
    mechanisms.  As compilers evolve, the underlying implementation can
    be changed with the macro definitions below.  FIXME: Update macros
    to use C++ standard attribute syntax when Octave moves to C++ 2011
    standard.  */
 
 #if defined (__GNUC__)
    /* The following attributes are used with gcc and clang compilers.  */
-#  define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__ (msg)))
+#  if __GNUC__ > 4 || (__GNUC__ == 4 && __GNUC_MINOR__ >= 5)
+#    define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__ (msg)))
+#  else
+#    define OCTAVE_DEPRECATED(msg) __attribute__ ((__deprecated__))
+#  endif
 #  define HAVE_OCTAVE_DEPRECATED_ATTR 1
 
 #  define OCTAVE_NORETURN __attribute__ ((__noreturn__))
 #  define HAVE_OCTAVE_NORETURN_ATTR 1
 
 #  define OCTAVE_UNUSED __attribute__ ((__unused__))
 #  define HAVE_OCTAVE_UNUSED_ATTR 1
 #else
